src/services/export-feature/export-feature-gpx.ts

```diff
@@ -0,0 +1,145 @@
+import { Feature } from 'ol'
+import { Geometry, LineString, Polygon } from 'ol/geom'
+import MultiLineString from 'ol/geom/MultiLineString'
+
+import { ExportFeature } from './export-feature'
+import { GPX } from './ol-format-gpx'
+
+export class ExportFeatureGpx extends ExportFeature {
+  /**
+   * Export a Gpx file
+   * @param features The features to export
+   * @param fileName The file name without the file extension (file extension ".gpx" will be added here)
+   * @param isTrack True if gpx should export tracks instead of routes (eg. use true for feature export, false for routing export)
+   */
+  export(features: Feature<Geometry>[], fileName: string, isTrack = false) {
+    const explodedFeatures = this.prepareFeatures(features, isTrack)
+    const content = this.generateContent(
+      explodedFeatures as Feature<Geometry>[],
+      fileName
+    )
+
+    this.download(fileName, content, 'gpx', 'application/gpx')
+  }
+
+  prepareFeatures(features: Feature<Geometry>[], isTrack = false) {
+    features.forEach(feature => {
+      const properties = feature.getProperties()
+
+      // NB. not sure if this is used in Lux
+      // LineString geometries, and tracks from MultiLineString
+      if ('feature_name' in properties) {
+        feature.set('name', properties['feature_name'], true)
+      }
+    })
+
+    let explodedFeatures = this.exploseFeatures(features)
+    if (isTrack) {
+      explodedFeatures = this.changeLineToMultiline(explodedFeatures)
+    } else {
+      explodedFeatures = this.changeMultilineToLine(explodedFeatures)
+    }
```

Comment:
Gpx files may contain Routes or Waypoints or Tracks. Here is the difference between Routes and tracks: https://hikingguy.com/how-to-hike/what-is-a-gpx-file/#routes
As for lux, I am not sure if **Routes** are used (I looked for `ctrl.exportGpx(true)` in the code), but I kept this condition just in case 🤔.

Reply:
It looks like Drawing export and Itinerary both use Tracks in v3, right?